python/cudf/cudf/utils/ioutils.py

```diff
@@ -1953,7 +1953,15 @@ def _fsspec_data_transfer(
 
     # Calculate total file size
     if file_like:
-        file_size = path_or_fob.size
+        try:
+            file_size = path_or_fob.size
+        except AttributeError:
+            # Find file size of there is no `size`
+            # attribute
+            old_file_position = path_or_fob.tell()
```

Comment:
Could we have a test where `old_file_position != 0`?

Reply:
Done